### PR TITLE
Add documentation for emacs configuration

### PR DESCRIPTION
**Pull request by timwoj**

### CODE REVIEW

**Comment by timwoj** [comment]
@ckreibich Any chance you could review this for accuracy? I set it all up in a bare emacs instance and it seemed to be working but it'd be good to have another set of eyes to make sure I didnt miss something.

**Comment by timwoj** [comment]
Christian also mentioned that it might be worth just merging this stuff into the zeek major-mode. I'll tinker with that.

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:22]: Is this actually a hard requirement, or just nice to have? Since this file isn't in the upstream package repos this seems to introduce some complications.

If we include it we need to make sure that formatting from `emacs-zeek-mode` does not interfere with formatting from the LSP server. Do you know whether this is disabled in `lsp-mode` by default (which would be fine)?

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:15]: I am not sure how familiar emacs users are with packages, but when I dropped below basic config into my system it failed (packages were not already present, and not installed automatically). When I installed them with `M-x package-install` things progressed.

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:45]: I am not sure how common it is to not install Zeek from packages. Maybe instead
```suggestion
1. If needed, add the path to `zeek-config`to your environment's `PATH` variable.

```

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:65]: I still seem to be missing a number of features I'd expect with this config. 

Works:
- when writing function calls I get a signature helper near the bottom, but I don't seem to be getting completion

Missing:
- no completions
- diagnostics missing: syntax errors in unsaved buffers not reported (should show diagnostics from tree-sitter parser); syntax errors in saved buffers not reported (should show diagnostics from `zeek --parse-only`)

This is with Homebrew emacs-27.2.

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:69]: Suggest to drop this as I suspect this is not a widespread use case.
```suggestion
```

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:66]: Could you add a pointer to `lsp-mode` (?) docs showing commands to get more information.

At least I have no understanding how to get access to other capabilities exposed by the server (hover, goto declaration/definition/implementation, list symbols, ...).

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:3]:
```suggestion
This file documents needed configuration to interface emacs with `zeek-language-server`. These instructions use `lsp-mode`. There's the possibility it would also work with `elgot` (an `lsp-mode` replacement), but that exercise is left to the reader. `lsp-mode` comes with an astonishing amount of options for configuring the look and operation of the LSP features in emacs. This guide only presents the most basic configuration.

```

File: emacs/README.md
Comment (bbannier) [inline emacs/README.md:1]:
```suggestion
# Emacs setup for zeek-language-server

```

Let's also link this from the top-level README.

Since this directory contains only a README file for now, I am wondering whether just adding a top-level `README.emacs` would be sufficient.

**Review by bbannier — CHANGES_REQUESTED** [review]
Thanks for writing this down @timwoj.

I opened a number of issues below which contain mostly questions.

Before merging somebody needs to fix the markdown for the linters running as part of `pre-commit`. I can do that, but have also added a CI step running that in case you want to look into it.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md:15]: Reworded the section a little bit to mention that they should all be installed before adding the snippet.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md:45]: Fixed.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md:64]: Dropped.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md:1]: I'd be alright with just moving it up.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md]: It is, because we need zeek-mode for the hook to lsp-mode to work.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md]: Completion depends on `company-mode`, which I didn't include in the guide (mostly because I haven't set it up myself yet). Diagnostics might be available in another buffer. Hit `ctrl-x right-arrow` to scroll between them.

File: emacs/README.md
Comment (timwoj) [reply emacs/README.md]: Added to the header.

File: emacs/README.md
Comment (bbannier) [reply emacs/README.md]: Just installing the `company` package got me completions, I think we should add it to the docs.

I couldn't find any diagnostics buffer, but `flymake-mode` seems to report something (`Flymake[0 0 1]` in modeline).

File: emacs/README.md
Comment (bbannier) [reply emacs/README.md]: > As for the formatting, I'm not sure. I'd have to play with it.

Looks like format-on-save is by default also disabled in `emacs-zeek-mode`, so the default config has no conflict. Unless behavior of one of the involved plugins change, there's probably no issue right now unless users go out of their way to mess things up.

> My only concern with doing that is that it's different from how most major modes handle things.

I am also wondering how that would impact users' ability to configure LSP integration. Like you write, this plugin comes with **a lot** of options, and users will likely want to tweak things.

**Comment by bbannier** [comment]
This looks like a great starting point to me and I'd be fine with just merging it (given some form of completion setup is documented). Would still be great if e.g., @ckreibich could take it for a test ride as well.

**Comment by bbannier** [comment]
I merged this as is in c1961f6f3ef835e536bcc5059a13ae87a4c735bd. Please feel free to amend if you find other useful bits you feel are worth documenting.

**Pull request closed without merging**
